### PR TITLE
Fix - Join/Leave Voice Event avoids Stage Channels

### PR DESCRIPTION
**Pull request by Lucasamiel0406**
**Please describe the changes this PR makes and why it should be merged:**
> A fix for Stage Channels crashing bots due to being `null` to Discord.JS, several people have been complaining about it a good time ago and even in nowadays.
Tested several times the event against false triggers or not triggering properly for voice channels and works fine.

**Status**

- [X] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [x] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by TheMonDon** [comment]
You're going to have to remove the semicolons before you can get this approved, alms linting thing hates semicolons.

I'll get danno or alm to start the eslint check when they're awake

**Review by TheMonDon — APPROVED** [review]

File: events/member_join_voice_channel_EVT.js [hunk 1]
```diff
@@ -11,8 +11,8 @@ module.exports = {
       if (!Bot.$evts['Member Join Voice Channel']) return
       const oldChannel = oldVoiceState.channel
       const newChannel = newVoiceState.channel
+      if (newChannel == null || !(!oldChannel && newChannel)) return
```
Comment (almeidx) [inline events/member_join_voice_channel_EVT.js:14]:
```suggestion
      if (oldChannel || !newChannel) return

```

File: events/member_leave_voice_channel_EVT.js [hunk 2]
```diff
@@ -7,13 +7,12 @@ module.exports = {
   mod (DBM) {
     DBM.Events = DBM.Events || {}
     const { Actions, Bot } = DBM
-
     DBM.Events.memberLeaveVoiceChannel = function (oldVoiceState, newVoiceState) {
       if (!Bot.$evts['Member Leave Voice Channel']) return
       const oldChannel = oldVoiceState.channel
       const newChannel = newVoiceState.channel
+      if (oldChannel == null || !(oldChannel && !newChannel)) return
```
Comment (almeidx) [inline events/member_leave_voice_channel_EVT.js:14]:
```suggestion
      if (!oldChannel || newChannel)) return

```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: events/member_join_voice_channel_EVT.js [hunk 3]
```diff
@@ -11,8 +11,9 @@ module.exports = {
       if (!Bot.$evts['Member Join Voice Channel']) return
       const oldChannel = oldVoiceState.channel
       const newChannel = newVoiceState.channel
+      if (oldChannel || !newChannel) return
+
       const server = (oldChannel || newChannel).guild
```
Comment (almeidx) [inline events/member_join_voice_channel_EVT.js:16]:
```suggestion
      const server = newChannel.guild

```

File: events/member_leave_voice_channel_EVT.js [hunk 4]
```diff
@@ -7,13 +7,13 @@ module.exports = {
...
     DBM.Events = DBM.Events || {}
     const { Actions, Bot } = DBM
-
     DBM.Events.memberLeaveVoiceChannel = function (oldVoiceState, newVoiceState) {
       if (!Bot.$evts['Member Leave Voice Channel']) return
       const oldChannel = oldVoiceState.channel
       const newChannel = newVoiceState.channel
+      if (!oldChannel || newChannel)) return
+
       const server = (oldChannel || newChannel).guild
```
Comment (almeidx) [inline events/member_leave_voice_channel_EVT.js:16]:
```suggestion
      const server = oldChannel.guild

```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: events/member_leave_voice_channel_EVT.js [hunk 5]
```diff
@@ -7,13 +7,13 @@ module.exports = {
   mod (DBM) {
     DBM.Events = DBM.Events || {}
     const { Actions, Bot } = DBM
-
     DBM.Events.memberLeaveVoiceChannel = function (oldVoiceState, newVoiceState) {
       if (!Bot.$evts['Member Leave Voice Channel']) return
       const oldChannel = oldVoiceState.channel
       const newChannel = newVoiceState.channel
+      if (!oldChannel || newChannel)) return
```
Comment (almeidx) [inline events/member_leave_voice_channel_EVT.js:14]:
```suggestion
      if (!oldChannel || newChannel) return

```

**Review by almeidx — APPROVED** [review]

**Pull request merged**